src/Microsoft.Health.Fhir.Liquid.Converter/Processors/JsonToHl7v2Processor.cs

```diff
@@ -0,0 +1,179 @@
+                foreach (var mapping in mappingsForSegment)
+                {
+                    var fieldPosition = mapping.Key;
+                    var fieldValue = mapping.Value?.ToString();
+                    var fieldPositionIndex = int.Parse(fieldPosition) - 1;
+
+                    if (!_segmentsWithFieldSeparator.Contains(segmentName))
+                    {
+                        fields[fieldPositionIndex] = new Hl7v2Field() { Value = fieldValue };
+                    }
+                    else
+                    {
+                        // the fields in segments with field separators should have a fieldPositionIndex of fieldPositionIndex - 1
+                        if (fieldPositionIndex >= 1)
+                        {
+                            fields[fieldPositionIndex - 1] = new Hl7v2Field() { Value = fieldValue };
+                        }
+
+                        if (fieldPositionIndex == fields.Count - 1)
+                        {
+                            fields.RemoveAt(fieldPositionIndex);
+                        }
+                    }
+                }
+
+                hl7Segments.Add(new Hl7v2Segment(string.Empty, fields) { SegmentName = segmentName });
+            }
+
+            return new Hl7v2Data() { Data = hl7Segments, EncodingCharacters = new Hl7v2EncodingCharacters() };
+        }
+
+        public string ConvertHl7MessageToString(Hl7v2Data message)
+        {
+            StringBuilder sb = new StringBuilder();
+            foreach (Hl7v2Segment segment in message.Data)
+            {
+                sb.Append(segment.SegmentName + message.EncodingCharacters.FieldSeparator);
+                foreach (Hl7v2Field field in segment.Fields)
+                {
+                    sb.Append(field.Value);
+                    sb.Append(message.EncodingCharacters.FieldSeparator);
+                }
+
+                sb.AppendLine();
+            }
+
+            return sb.ToString();
+        }
+
+        protected override Context CreateContext(ITemplateProvider templateProvider, IDictionary<string, object> data)
+        {
+            // Load data and templates
+            var cancellationToken = Settings.TimeOut > 0 ? new CancellationTokenSource(Settings.TimeOut).Token : CancellationToken.None;
+            var context = new JSchemaContext(
+                environments: new List<Hash> { Hash.FromDictionary(data) },
+                outerScope: new Hash(),
+                registers: Hash.FromDictionary(new Dictionary<string, object> { { "file_system", templateProvider.GetTemplateFileSystem() } }),
+                errorsOutputMode: ErrorsOutputMode.Rethrow,
+                maxIterations: Settings.MaxIterations,
+                formatProvider: CultureInfo.InvariantCulture,
+                cancellationToken: cancellationToken)
+            {
+                ValidateSchemas = new List<JsonSchema>(),
```

Comment:
Just out of curiosity, I've not yet figured out what this ( ValidateSchemas = new List<JsonSchema>(),) is needed for, I 've only seen this be set for the ToJsonProcessor and was wondering if you happen to know more about this.

Reply:
I don't know much about it. it seems like the JSON processors support some kind of schema validation on the output but I am not too familiar with it: https://github.com/microsoft/FHIR-Converter/tree/b06f905598b1cda9abc0818f6d9c7331aa5657fe/src/Microsoft.Health.Fhir.Liquid.Converter.UnitTests/TestData/TestTemplates/ValidValidateTemplates